### PR TITLE
[TT-7126] add authority header to grpc

### PR DESCRIPTION
**Pull request by titpetric**
## Related Issue (if applicable)

https://tyktech.atlassian.net/browse/TT-7126 (master, 4-lts, 5-lts)

- The `grpc_authority` option had been added. This updates the relevant tutorial to document it.

Before: https://tyk.io/docs/plugins/supported-languages/rich-plugins/grpc/tutorial-add-grpc-plugin-api/
After: https://deploy-preview-2532--tyk-docs.netlify.app/nightly/docs/plugins/supported-languages/rich-plugins/grpc/tutorial-add-grpc-plugin-api/

## Description

Additional improvements to the docs have been made here:

- Improved the leading code block to be a `diff` (don't overload the user with a huge json snippet out of the gate)

image

- Excluded `.language-diff` from having a "copy to clipboard" button (no need to copy/paste illustrative diffs)

- General content update for readability, bullet lists, headings, paragraphs.

## Motivation and Context

Base doc didn't meet my standards, edited to improve.

## Types of changes

- [ ] Fixing typo (please merge to production)
- [x] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I have added a preview link.

### CODE REVIEW

**Comment by letzya** [comment]
@titpetric get a peer to review the content please

**Comment by letzya** [comment]
fyi @titpetric not sure about the "diff" thing - it fits release notes more than regular docs since in a year it won't be a new addition

**Comment by titpetric** [comment]
> fyi @titpetric not sure about the "diff" thing - it fits release notes more than regular docs since in a year it won't be a new addition

It's a partial, similar to the one below. Would just leaving the value to be added be better?

I'll give it a quick poll, reached out for a peer review.

File: tyk-docs/content/plugins/supported-languages/rich-plugins/grpc/tutorial-add-grpc-plugin-api.md [hunk 1]
```diff
@@ -11,45 +11,18 @@ aliases:
...
-    "strip_listen_path": true
-  },
-  "custom_middleware_bundle": "test-bundle"
+   "name": "Tyk Test API",
+   ...
++  "custom_middleware_bundle": "test-bundle"
 }
 ```
 
+The value of the field will be used in combination with the gateway settings to construct a bundle URL.
```
Comment (sredxny) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/grpc/tutorial-add-grpc-plugin-api.md:24]: would be nice to know which settings exactly are being used to construct the url and understand exactly how the url is built, so customers will know small details such as: should this value start with slash? should this be a valid url? should be a path?

**Review by sredxny — COMMENTED** [review]
overall looks good to me. Just added a comment o something we can discuss

**Review by sredxny — APPROVED** [review]
LGTM ✅

**Comment by letzya** [comment]
@sredxny @titpetric up to you - it can be auto-merge if you resolve the comment

**Comment by titpetric** [comment]
@letzya If Sophie gets back to me (on vacation) with her non-technical POV, we'll open up an additional improvement ticket. Sredny comment is resolved as the docs explains how the value is used below the comment. Happy to get it merged :+1:

**Pull request merged**